enhancements/microshift/microshift-supportability-tools.md

```diff
@@ -0,0 +1,365 @@
+the relevant data from the different components and applications in a system.
+
+By executing `sos report` it will generate a detailed report with all the
+enabled plugins (listed by running `sos report -l`).
+
+A [plugin](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin) is a
+python script included as part of `sos` and makes the tool extensible. Plugins
+also have an enablement check to run by themselves upon a report command, or
+may be invoked manually.
+
+Among the plugins already available we can find one for [OpenShift](https://github.com/sosreport/sos/blob/main/sos/report/plugins/openshift.py).
+This plugin relies on the usage of `oc` command line tool and it gathers data
+from OpenShift API resources, collecting logs for pods and kubelet. While
+MicroShift shares similarities with this plugin it is not directly usable, as
+the APIs are not the same, resources are not the same, and the output format
+is required to follow `must-gather` format.
+
+Since none of the available plugins fulfills MicroShift's needs, we will need a
+new one.
+
+Requirements for the `sos` plugin follow:
+* Not require usage of other tools than what is already present in the host.
+  No additional images or componentes shall be pulled when performing a report.
+* Have `must-gather` format to use already existing debug tools.
+* Retrieve MicroShift specific resources only, not covered by other plugins.
+  These include cluster resources, MicroShift configuration, MicroShift
+  version and MicroShift logs. Some specifics, like ovnk will require an
+  additional plugin, as openshift does. Other components, such as CRI-O, have
+  their own plugins.
+* Provide a single command pre-configured for full report gathering.
+  In order to have a single and simple command to gather the report, provide a
+  profile configuration file for `sos` to run all relevant plugins in one shot.
+* Must not leak sensitive data: IPs, hostnames, passwords, certificates, etc.
+  The /var/lib/microshift directory holds certificates and etcd data, therefore
+  it must be skipped.
+
+Existing debugging tools in use by CEE include [omg](https://github.com/kxr/o-must-gather)
+and [omc](https://github.com/gmeghnag/omc). These provide a CLI like experience
+in terms of getting information from a live cluster, ressembling `oc` commands,
+but coming from a must-gather report. Since they were born out of must-gather,
+they expect the same file structure layout. The must-gather file system
+hierarchy is driven by `oc adm inspect` commands, gathering cluster resources
+data that these tools read upon later.
+
+To achieve the same results the most simple approach is to include the `oc`
+binary with MicroShift as a dependency. The plugin can then call
+`oc adm inspect` to produce the output required. There is no API between the
+plugin and MicroShift, easing maintainability and allowing enough flexibility
+for the future.
```

Comment:
What about `sos`, is it included by default or it should be mentioned here that it also needs to be included just like oc?

Reply:
We added it to MicroShift as a dependency until it is included by default in the R4E base layer.